plugins/woocommerce/src/StoreApi/Utilities/OrderController.php

```diff
@@ -171,7 +171,26 @@ public function validate_order_before_payment( \WC_Order $order ) {
 		$this->validate_coupons( $order );
 		$this->validate_email( $order );
 		$this->validate_selected_shipping_methods( $needs_shipping, $chosen_shipping_methods );
-		$this->validate_addresses( $order );
+		$this->validate_addresses( $order, $needs_shipping );
+	}
+
+	/**
+	 * Final validation for existing orders, ran before payment is taken.
+	 *
+	 * By this point we have an order populated with customer data and items.
+	 *
+	 * Since the cart is not involved, we don't validate shipping methods and assume the order already
+	 * contains the correct shipping items.
+	 *
+	 * @throws RouteException Exception if invalid data is detected.
+	 * @param \WC_Order $order Order object.
+	 */
+	public function validate_existing_order_before_payment( \WC_Order $order ) {
+		$needs_shipping = $order->needs_shipping();
+
+		$this->validate_coupons( $order, true );
+		$this->validate_email( $order );
+		$this->validate_addresses( $order, $needs_shipping );
```

Comment:
As mentioned in the PR description, I'm skipping the `validate_selected_shipping_methods` method since `WC()->shipping()->get_packages()` returns an empty array when used with an existing order instead of a cart.

Reply:
Yes after the order is placed there is no reference back to them, nothing to validate